flang/lib/Evaluate/intrinsics.cpp

```diff
@@ -2333,6 +2343,11 @@ std::optional<SpecificCall> IntrinsicInterface::Match(
       CHECK(FloatingType.test(*category));
       resultType = DynamicType{*category, defaults.doublePrecisionKind()};
       break;
+    case KindCode::quadPrecision:
+      CHECK(result.categorySet == CategorySet{*category});
+      CHECK(FloatingType.test(*category));
+      resultType = DynamicType{*category, defaults.quadPrecisionKind()};
```

Comment:
Quad precision is not supported on all targets.  You need to catch them and report an error.